src/NzbDrone.Core/Indexers/IndexerDefinition.cs

```diff
@@ -10,6 +10,7 @@ public class IndexerDefinition : ProviderDefinition
         public DownloadProtocol Protocol { get; set; }
         public bool SupportsRss { get; set; }
         public bool SupportsSearch { get; set; }
+        public int Priority { get; set; } = 100;
```

Comment:
You have two usecases that are competing for a particular default: 
- Users with fallback indexer with lower grab-limits or generally more junk releases.
- Users with premium indexers that they prefer.

The DownloadClients priority was more of the Fallback variety and thus a default of 1 (and 1 hidden on the UI), 2 being the fallback.

I don't like the idea of an arbitrarily large 100 number. I also don't want to see it on the UI unless the user is actually using priorities, and definitely not a "Priority: 100" on every indexer by default.

I see two approaches:
- We ignore the premium indexer usecase, anyone wanting it can simply set all others to a higher number. Default would be 1, only values > 1 shown on the ui.
- We switch to 0 as default and allow +1 (higher prio) and -1 (lower prio). Then we can hide 0 on the ui and won't have an arbitrary default. This also means that DownloadClient prio should be updated to that scheme as well for consistency (and a sql in the migration `UPDATE DownloadClients SET Priority = (1 - Priority)`)

Reply:
Well I actually became interested in working on this because of my case that differs.
I personally much rather download from my junk indexers first before my premium ones. I am trying to build my ratios for my more premium indexers and would rather use them as a last resort. This is ok by me because sonarr has so many filters for junk indexers that it is rarely a issue that it gets it wrong.

If we are comparing it to Download Client Priority, that is a 1-50 range option. 1 being default. So it defaults a bit differently but that is because normally you don't have more then one, that might be a assumption on my part but I don't think there and many users with more then one. Indexers is where I see people having a lot. I personally have 15 set up and plan on adding more as my private tracker collection grows.

Now the reason I think priority should be shown for each indexer is because we are now sorting the the cards by this value so I think it should always be shown. I do see it wouldn't be needed for people not using it so I do agree that we could hide it if all are set to 100. 

I personally don't like the -1 +1 approach. When talking about the priority the easiest thing to understand is using it to describe positive numbers. I think you would get more questions if you try introducing negatives here.  

I understand that setting the default priority to the lowest available differs from the Download client priority but I do think this is the best way to handle this. Another argument for this way is if someone set priorities, went to create a new one they would need to remember they need to set the priority for the new indexer. Defaulting to the highest I think would just cause more issues. You should have to purposely set something for it to be considered a higher priority not default into it. 

I would be much more inclined to set the max priority to 50 to mach the download client and keep the current implementation.

Reply:
> So it defaults a bit differently but that is because normally you don't have more then one, that might be a assumption on my part but I don't think there and many users with more then one.

I think that's a fair assumption, 1 for torrents and 1 for newsgroups.

> Now the reason I think priority should be shown for each indexer is because we are now sorting the the cards by this value so I think it should always be shown.

I still don't think sorting by priority in this list makes sense, it'd make sense in a dragable list, but less so in the card layout used.

> I do see it wouldn't be needed for people not using it so I do agree that we could hide it if all are set to 100.

I'm in favour of this as well, if you're not using priority at all it's just noise.

> Defaulting to the highest I think would just cause more issues. You should have to purposely set something for it to be considered a higher priority not default into it.

Same thing for defaulting to the lowest, if you have 10 indexers and want one to have a lower priority then you're forced to edit every indexer to fix that. A default of either the highest or the lowest doesn't make sense to me, set it right in the middle and following download clients it should be 25.


TLDR:

- Range from 1-50
- Default priority to 25 (users can go higher and lower without editing every other indexer)
- Hide priority if they're all the default
- Don't sort by priority, just name

Reply:
> Same thing for defaulting to the lowest, if you have 10 indexers and want one to have a lower priority then you're forced to edit every indexer to fix that. A default of either the highest or the lowest doesn't make sense to me, set it right in the middle and following download clients it should be 25.

Fair point, I could get behind that.

I'll work on fixing this up based on you're comments